scipy/io/wavfile.py

```diff
@@ -243,7 +244,15 @@ def read(filename, mmap=False):
             chunk_id = fid.read(4)
 
             if not chunk_id:
-                raise ValueError("Unexpected end of file.")
+                if data_chunk_received:
+                    # End of file but data successfully read
+                    warnings.warn(
+                        "Reached EOF (filesize reported as {:d} by header)."
```

Comment:
How about "Reached EOF prematurely, read only %d frames of the %d expected from the header" or so

Reply:
Updated - I'm not completely sure of your use of "frames" - would this be calculated by the shape of `data`? Then how would the "expected" frames be calculated? Dividing expected length of the data chunk (filesize minus all the other chunks) by the number of channels?  Or am I just overthinking this and it's just equivalent to "bytes"?